samples/nrf9160/pdn/src/main.c

```diff
@@ -10,6 +10,11 @@
 #include <modem/pdn.h>
 #include <modem/lte_lc.h>
 #include <nrf_modem_at.h>
+#include <nrf_socket.h>
+
+#define SIN(addr)  ((struct nrf_sockaddr_in *)(addr))
```

Comment:
Maybe the existing inline functions net_sin6() and net_sin() already do this? (From https://docs.zephyrproject.org/apidoc/latest/group__ip__4__6.html)

Reply:
There is a difference between the `nrf_sockaddr_in` and `sockaddr_in` structs, so we need to use the `nrf_sockaddr_in` struct. Similarly for IPv6.

```
struct sockaddr_in {
	sa_family_t sin_family;    /* AF_INET      */
	uint16_t sin_port;         /* Port number  */
	struct in_addr sin_addr;   /* IPv4 address */
};

struct nrf_sockaddr_in {
	uint8_t sin_len;                /** Length of this data structure. */
	nrf_sa_family_t sin_family;	/** Socket family. */
	nrf_in_port_t sin_port;	        /** Port, in network byte order. */
	struct nrf_in_addr sin_addr;	/** IPv4 address. */
};
```

Reply:
Ah, okay. That makes sense. I was kind of thinking the nrf_ prefixes were just an alias (or the reverse, i.e. without was an alias for nrf_).